test/homework2_test.exs

```diff
@@ -0,0 +1,14 @@
+defmodule Homework2Test do
+  use ExUnit.Case
+  doctest Homework2
+
+  test "it responds to a ping with a pong" do
+    pid = Homework2.send({:ping, self()})
+    assert_receive {:pong, ^pid}
+  end
+
+  test "it responds to a different message with an error" do
+    pid = Homework2.send({:pong, self()})
```

Comment:
How nice of the caller sending a malformed message to still provide `pid` here :)